mutation/mutation.cc

```diff
@@ -194,6 +196,71 @@ mutation reverse(mutation mut) {
     return *std::move(mut).consume(reverse_rebuilder, consume_in_reverse::yes).result;
 }
 
+future<> split_mutation(mutation source, std::vector<mutation>& target, size_t max_size) {
+    class splitter {
+        const schema_ptr _schema;
+        std::vector<mutation>& _target;
+        std::optional<mutation_rebuilder_v2> _mut_builder;
+        size_t _size = 0;
+        const size_t _max_size;
+        void maybe_flush() {
+            if (_size >= _max_size) {
+                _target.emplace_back(_mut_builder->flush());
+            }
+        }
+    public:
+        splitter(schema_ptr schema, std::vector<mutation>& target, size_t max_size)
+            : _schema(std::move(schema))
+            , _target(target)
+            , _max_size(max_size)
+        {
+        }
+        void consume_new_partition(const dht::decorated_key& dk) {
+            _mut_builder.emplace(_schema);
+            _size += _mut_builder->consume_new_partition(dk).memory_usage(*_schema);
+        }
+        void consume(tombstone t) {
+            _mut_builder->consume(t);
+        }
+        stop_iteration consume(static_row&& sr) {
+            _size += sr.memory_usage(*_schema);
+            _mut_builder->consume(std::move(sr));
+            maybe_flush();
+            return stop_iteration::no;
+        }
+        stop_iteration consume(clustering_row&& cr) {
+            _size += cr.memory_usage(*_schema);
+            _mut_builder->consume(std::move(cr));
+            maybe_flush();
```

Comment:
So we can open a partition, then flush, then consume partition end? How does that work?

Reply:
Yeah I know -- but how does that address my question?

In fact good that you referenced the `flush()` definition, which has a comment above it:
```cpp
    // Might only be called between consume_new_partition()
    // and consume_end_of_partition().
```
so my suspicions appear to be valid. The code IIUC is `flush()`ing in the middle of a partition, which is incorrect.

Reply:
@denesb maybe we should have checks in debug mode that verify such preconditions in `mutation_rebuilder{_v2}`

Reply:
I don't know what you are talking about. We _are_ between `consume_new_partition` and `consume_end_of_partition`.

Reply:
Ok (－‸ლ) I misunderstood the code and the comment. Indeed it makes sense that we flush in the middle of a partition and not in between partitions, that's the whole point